<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>TopMSV</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="">

  <link href="lib/bootstrap.css" rel="stylesheet">
  <link href="lib/bootstrap-responsive.css" rel="stylesheet">
  <link href="public/css/style.css" rel="stylesheet">
  <link rel="shortcut icon" href="toppic.ico">
</head>

<body style="overflow-y:scroll">

<div class="container">

  <div class="masthead">
    <h2 class="muted"><font color=#0066DD>TopMSV</font></h2>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container">
          <ul class="nav">
            <li><a href="index.html">Home</a></li>
            <li><a href="download.html">Download</a></li>
            <li><a href="manual.html">Manual</a></li>
            <li><a href="tutorial.html">Tutorial</a></li>
            <li><a href="contact.html">Contact</a></li>
          </ul>
        </div>
      </div>
    </div><!-- /.navbar -->
  </div><!-- /.masthead -->

  <div class="row-fluid" style="font-size:16px">
    <div class="span-12">
      <p> In this tutorial, we will show how to use the TopMSV Windows server to
      visualize an LC-MS/MS data set.
      </p>
      <h3>1. Download the TopMSV server</h3>
      <ol class="textList">
        <li>Go to the <a href="http://toppic.org/software/topmsv/index.html">Download</a> page.</li>
        <li>Choose the download type "Windows 64-bit zip file." </li>
        <li>Fill out the registration form, and click "I accept license
          agreement and download TopMSV" to download TopMSV.</li> 
        <li>Create a new folder <code>C:\topmsv\server</code>,
          save the zip file to the folder, and unzip the
          downloaded zip file to the folder
        <code>C:\topmsv\server</code>.</li>
      </ol>
      <h3>2. Start the TopMSV server</h3>
        <p>Double click the file <code>startServer.bat</code> in the folder
        <code>C:\topmsv\server</code> to start the server.
        </p>
      <h3>3. LC-MS/MS data visualization </h3>
        <h4 class="sectionTitle">3.1 Top-down MS/MS Dataset</h4>
        <p>
          In the MS experiment, the protein extract of S. typhimurium was
          reduced with dithiothreitol and alkylated with iodoacetamide.  The
          protein mixture was first separated by gas-phase fractionation,
          resulting in 7 fractions.  Each fraction was separated by an HPLC
          system coupled to an LTQ-Orbitrap mass spectrometer (Thermo Fisher
          Scientific).  MS and MS/MS spectra were collected at a resolution of
          60,000 and 30,000, respectively. In this tutorial, we use only the
          LC-MS/MS data file of the first fraction. The raw file was converted
          to an mzML file (st_1.mzML) using msconvert in ProteoWizard.
        </p>
        <p>Click <a
                   href="filedownload.php?filename=./tutorial_data/st_1.mzML">here</a>
        to download the data set, save it into a local folder
        <code>C:\topmsv\data</code>, and unzip it.</p>

        <h4 class="sectionTitle">3.2 Protein sequence database</h4>
        <p>A S. typhimurium <a href="https://www.uniprot.org/uniprot/?query=proteome:UP000001014%20reviewed:yes">proteome database</a>
          of 1,799 proteins was downloaded from the UniProt database.
        </p>
  
        <p>Click <a href="filedownload.php?filename=./tutorial_data/uniprot-st.fasta">here</a>
          to download the protein database and save it into the folder 
          <code>C:\topmsv\data</code>.</p>
  
        <h4 class="sectionTitle">3.3 Upload data</h4>
        <ol class="textList">
          <li>Open the Google Chrome web browser, type in the web server
            address <code>http://localhost:: </code> in the address bar, and 
          click the "Enter" button to go the main webpage of the TopMSV server.</li>
          <li>Click the "Sign in" button in the top right corner.</li>
          <p><img class="small-image" src="public/images/index_1.PNG" style="border: #00008B 1px solid;"/></p>
          <li>Click "Upload" in the navigation bar.</li>
          <p><img class="small-image" src="public/images/index_2.PNG" style="border: #00008B 1px solid;"/></p>
          <li>Type "st_1" as the project name, click the "Select mzML file"
            button to select the data file
            <code>C:\topmsv\data\st_1.mzML</code>,
            and click the "Submit" button to update the data file.</li>
          <p><img class="small-image" src="public/images/upload_1.PNG" style="border: #00008B 1px solid;"/></p>
        </ol>
      
        <h4 class="sectionTitle">3.4 Data visualization</h4>
        <ol class="textList">
          <li>Click "My project" in the project in the table in the main page. </li>
          <p><img class="small-image" src="public/images/index_3.PNG" style="border: #00008B 1px solid;"/></p>
          <li>Click "Link" in the project in the table in the main page.</li>
          <p><img class="small-image" src="public/images/project_1.PNG" style="border: #00008B 1px solid;"/></p>
        </ol>
          <p>Information on how to interact with each visualization can be found at <a href="manual.html">manual</a> page.</p>

          <h4 class="sectionTitle">3.5 Add deconvolution results</h4> 
          <p>Annotation on isotopic envelopes can be added to the spectra graphs by running TopFD or manually uploading msalign files from local drive. </p>
          <h4 class="sectionTitle">3.5.1 Run TopFD</h4> 
          <ol class="textList">
            <li>Click "TopFD" in the top right corner of the visualization page.</li>
            <p><img class="small-image" src="public/images/topfd_1.png" style="border: #00008B 1px solid;"/></p>
            <li>Make changes to the parameters if needed, and click "submit" button.</li>
            <p><img class="small-image" src="public/images/topfd_2.PNG" style="border: #00008B 1px solid;"/></p>
          </ol>
          <h4 class="sectionTitle">3.5.2 Upload deconvolution results</h4> 
          <ol class="textList">
            <li>Click "Upload" in the top right corner of the visualization page.</li>
            <p><img class="small-image" src="public/images/manual_1.png" style="border: #00008B 1px solid;"/></p>
            <li>Click "Choose file" to locate ms1 and ms2 msalign files generated by TopFD, and click "upload" button.</li>
            <p><img class="small-image" src="public/images/msalign_1.PNG" style="border: #00008B 1px solid;"/></p>
            <li>After envelope data is processed, isotopic envelope annotations are added to the spectra graphs. A table containing the envelope data is also generated.</li>
            <p><img class="small-image" src="public/images/msalign_2.PNG" style="border: #00008B 1px solid;"/></p>
          </ol>

          <h4 class="sectionTitle">3.6 Upload feature data</h4> 
          <ol class="textList">
            <li>Click "Upload" in the top right corner of the visualization page.</li>
            <p><img class="small-image" src="public/images/manual_1.png" style="border: #00008B 1px solid;"/></p>
            <li>Click "Choose file" to locate a mzrt csv file generated by TopFD, and click "upload" button.</li>
            <p><img class="small-image" src="public/images/mzrt_1.PNG" style="border: #00008B 1px solid;"/></p>
            <li>After mzrt data is processed, annotations for each feature (dotted rectangles) are added to the 3d graph. When a mouse cursor hovers over a rectangle, a tooltip is displayed with the current feature information.</li>
            <p><img class="small-image" src="public/images/mzrt_2.PNG" style="border: #00008B 1px solid;"/></p>
          </ol>

          <h4 class="sectionTitle">3.7 Upload PrSM identification data</h4> 
          <p>PrSM identification data can be uploaded by running TopPIC or manually uploading identification file from local drive. 
            When the identification data is added, it is possible to adjust input data such as peaks, ions, PTMs and redraw spectra graphs.</p>
          <h4 class="sectionTitle">3.7.1 Run TopPIC</h4> 
          <ol class="textList">
            <li>Click "TopPIC" in the top right corner of the visualization page.</li>
            <p><img class="small-image" src="public/images/toppic_1.png" style="border: #00008B 1px solid;"/></p>
            <li>Upload a protein database file from <code>/var/opt/topmsv</code> by clicking "Choose file."</li>
            <li>(Optional) When uploading a text file for fixed PTMs, follow the format in this <a href="tutorial_data/fixed_mods.txt">sample file</a>. When uploading a text file for PTM characterization, follow the format in this <a href="tutorial_data/common_mods.txt">sample file</a>.</li>
            <p><img class="small-image" src="public/images/toppic_2.png" style="border: #00008B 1px solid;"/></p>
            <li>Make changes to the parameters if needed, and click "submit" button.</li>
          </ol>
          <h4 class="sectionTitle">3.7.2 Manual upload</h4> 
          <ol class="textList">
            <li>Click "Upload" in the top right corner of the visualization page.</li>
            <p><img class="small-image" src="public/images/manual_1.png" style="border: #00008B 1px solid;"/></p>
            <li>Click "Choose file" to locate the protein identification file generated by TopPIC, and click "upload" button.</li>
            <p><img class="small-image" src="public/images/prsm_1.PNG" style="border: #00008B 1px solid;"/></p>
          </ol>
          <h4 class="sectionTitle">3.8 Inspect page</h4> 
          <ol class="textList">
            <li>After running TopPIC, click "Inspect" button from the envelope table.</li>
            <p><img class="small-image" src="public/images/inspect_1.PNG" style="border: #00008B 1px solid;"/></p>
            <li>Input data are pre-filled using the data from PrSM identification file. Make any changes if necessary, then click "Submit" to display visualizations.</li>
            <p><img class="small-image" src="public/images/inspect_2.PNG" style="border: #00008B 1px solid;"/></p>
            <li>A protein sequence with a MS/MS spectra graph, a mono mass graph, a mono mass table, and a theoretical fragment ion mass table appear.</li>
            <p><img class="small-image" src="public/images/inspect_3.PNG" style="border: #00008B 1px solid;"/></p>
          </ol>
          <p>Information on how to interact with each visualization can be found at <a href="manual.html">manual</a> page.</p>
          <hr>
</ol>
    </div>
  </div>
</div>

<script src="lib/jquery-1.7.1.min.js"></script>
<script>
    //adding a button to copy command line commands
    function copyToClipboard(cmdId){
        const copyText = document.getElementById(cmdId).textContent;
        const textArea = document.createElement('textarea');

        textArea.textContent = copyText;
        document.body.append(textArea);
        textArea.select();
        document.execCommand("copy");

        //show tooltip that shows the text is copied
        initializeTooltip(cmdId);

        //remove the text area
        removeTextArea(textArea);

    }
    function removeTextArea(textArea){//removing the textarea that was created to just for copying purpose
        textArea.remove();
    }

    function initializeTooltip(cmdId){
      const tooltip = document.createElement('div');

      document.getElementById(cmdId + "Container").appendChild(tooltip);
      tooltip.classList.add("copyBtnTooltip");
      tooltip.textContent = "Copied to clipboard";

      setTimeout(function resetToolTip(){//give time for a tooltip to appear and then completely disappear
        tooltip.style.opacity = "0";
        tooltip.style.transition = "opacity 1s";

        setTimeout(function removeToolTip(){
                tooltip.remove();
        }, 500);
      }, 200);
    }
</script>
</body>
</html>
