<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>TopMSV</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="">

  <link href="lib/bootstrap.css" rel="stylesheet">
  <link href="lib/bootstrap-responsive.css" rel="stylesheet">
  <link href="public/css/style.css" rel="stylesheet">
  <link rel="shortcut icon" href="toppic.ico">
</head>

<body style="overflow-y:scroll">

<div class="container">

  <div class="masthead">
    <h2 class="muted"><font color=#0066DD>TopMSV</font></h2>
    <div class="navbar">
      <div class="navbar-inner">
        <div class="container">
          <ul class="nav">
            <li><a href="index.html">Home</a></li>
            <li><a href="download.html">Download</a></li>
            <li><a href="manual.html">Manual</a></li>
            <li><a href="tutorial.html">Tutorial</a></li>
            <li><a href="contact.html">Contact</a></li>
          </ul>
        </div>
      </div>
    </div><!-- /.navbar -->
  </div><!-- /.masthead -->

  <div class="row-fluid" style="font-size:16px">
    <div class="span-12">
      <p> In this tutorial, we will show how to use the TopMSV Windows server to
      visualize an LC-MS/MS data set.
      </p>
      <h3>1. Downloading TopMSV</h3>
      <ol class="textList">
        <li>Go to the 
          <a href="http://toppic.org/software/topmsv/index.html">Download</a>
          web page.</li>
        <li>Choose the download type "Windows 64-bit zip file." </li>
        <li>Fill out the registration form, and click "I accept license
          agreement and download TopMSV" to download TopMSV.</li> 
        <li>Create a new folder <code>C:\topmsv\server</code>,
          save the zip file to the folder, and unzip the
          downloaded zip file to the folder
        <code>C:\topmsv\server</code>.</li>
      </ol>
      <h3>2. Starting a TopMSV server</h3>
        <p>Double click the batch file <code>startServer.bat</code> in the folder
        <code>C:\topmsv\server</code> to start the server.
        </p>
      <h3>3. LC-MS/MS data visualization </h3>
        <h4 class="sectionTitle">3.1 Data</h4>
        <h4 class="sectionTitle">3.1.1 A top-down MS/MS Data set</h4>
        <p>
          In the MS experiment, the protein extract of S. typhimurium was
          reduced with dithiothreitol and alkylated with iodoacetamide.  The
          protein mixture was first separated by gas-phase fractionation,
          resulting in 7 fractions.  Each fraction was separated by an HPLC
          system coupled to an LTQ-Orbitrap mass spectrometer (Thermo Fisher
          Scientific).  MS and MS/MS spectra were collected at a resolution of
          60,000 and 30,000, respectively. In this tutorial, we use only the
          LC-MS/MS data file of the first fraction. The raw file was converted
          to an mzML file (st_1.mzML) using msconvert in ProteoWizard.
        </p>
        <p>Click <a
                   href="filedownload.php?filename=./tutorial_data/st_1.mzML">here</a>
        to download the data set, save it into the folder
        <code>C:\topmsv\data</code>, and unzip it.</p>

        <h4 class="sectionTitle">3.1.2 A protein sequence database</h4>
        <p>A S. typhimurium <a href="https://www.uniprot.org/uniprot/?query=proteome:UP000001014%20reviewed:yes">proteome database</a>
          of 1,799 proteins was downloaded from the UniProt database.
        </p>
  
        <p>Click <a href="filedownload.php?filename=./tutorial_data/uniprot-st.fasta">here</a>
          to download the protein database and save it into the folder 
          <code>C:\topmsv\data</code>.</p>

        <h4 class="sectionTitle">3.1.3 A text file of common post-translational
          modifications (PTMs)</h4>

        <p>Click <a
                   href="filedownload.php?filename=./tutorial_data/common_mods.txt">here</a>
          to download a text file containing common PTMs and save it into the folder 
          <code>C:\topmsv\data</code>.
        </p>
  
        <h4 class="sectionTitle">3.2 Uploading data</h4>
        <ol class="textList">
          <li>Open the Google Chrome web browser, type in the web server address
            <code>http://localhost:8443/</code> in the address bar, and click
            the "Enter" button in the keyboard to go to the main webpage of the
          TopMSV server.</li>
          <li>Click the "Sign in" button in the top right corner to sign in as a
            guest.</li>
          <p><img class="small-image" src="public/images/index_1.PNG" style="border: #00008B 1px solid;"/></p>
          <li>Click the "Upload" tab.</li>
          <p><img class="small-image" src="public/images/index_2.PNG" style="border: #00008B 1px solid;"/></p>
          <li>Type in "st_1" as the project name, click the "Select mzML file"
            button to select the data file
            <code>C:\topmsv\data\st_1.mzML</code>,
            and click the "Submit" button to update the data file.</li>
          <p><img class="small-image" src="public/images/upload_1.PNG" style="border: #00008B 1px solid;"/></p>
        </ol>
      
        <h4 class="sectionTitle">3.3 Data visualization</h4>
        <ol class="textList">
          <li>Click the "My project" tab to show all projects.  
          </li>
          <p><img class="small-image" src="public/images/index_3.PNG" style="border: #00008B 1px solid;"/></p>
          <li>Click "Link" in the "Project link" column to visualize the
            LC-MS/MS data. A manual for data visualization can be found at the 
            <a href="manual.html">manual</a> webpage.
            <font style="color:red">After the data file is uploaded, it takes several
              minutes for the server to store peak information into a database. if
              the "Project link" column shows "processing," please wait for several
              minutes until a link is available.
            </font>
          </li>
          <p><img class="small-image" src="public/images/project_1.PNG" style="border: #00008B 1px solid;"/></p>
        </ol>

        <h4 class="sectionTitle">3.4 Adding deconvolution results</h4> 
        <p>Spectral deconvolution results can be added to the LC-MS/MS data by
        (1) running TopFD at the server or (2) runing TopFD locally and
        uploading deconvolution result files to the server. When spectral
        deconvolution results are available, theoretical isotopic envelopes will
        be shown in 2D views of MS1 and MS/MS spectra.</p>
        <h4 class="sectionTitle">3.4.1 Running TopFD at the server</h4> 
        <ol class="textList">
          <li>Click the "TopFD" button in the top right corner of the
            visualization webpage.</li>
          <p><img class="small-image" src="public/images/topfd_1_part.png" style="border: #00008B 1px solid;"/></p>
          <li>Select parameter settings of TopFD and and click the "Submit"
            button. A manual of parameter settings of TopFD can be found 
            <a href="https://www.toppic.org/software/toppic/manual.html">here</a>.
          </li>
          <p><img class="small-image" src="public/images/topfd_2.PNG" style="border: #00008B 1px solid;"/></p>
          <li>After the task is submitted, the progress of the task can be found
            in the "Tasks" tab. When the task is finished, spectral
            deconvolution results reported by TopFD will be added to the
            database of the data set. 
          </li>
        </ol>
        <h4 class="sectionTitle">3.4.2 Uploading deconvolution results</h4> 
        <ol class="textList">
          <li>Run TopFD to perform spectral deconvolution of the data file
            st_1.mzML. A tutorial for runing TopFD can be found 
            <a href="https://www.toppic.org/software/toppic/tutorial.html">here</a>.
            Spectral deconvolution result files reported by TopFD include:
            <ul>
              <li>C:\topmsv\data\st_1_file\st_1_ms1.msalign: deconvolution results of MS1
                spectra. </li>
              <li>C:\topmsv\data\st_1_ms2.msalign: deconvolution results of MS/MS spectra. </li>
              <li>C:\topmsv\data\st_1_file\st_1_frac.mzrt.csv: proteoform features in the LC-MS
                map.</li>
            </ul>
          </li>  
          <li>Click the "Upload" button in the top right corner of the
            visualization webpage.</li>
          <p><img class="small-image" src="public/images/manual_1_part.png" style="border: #00008B 1px solid;"/></p>
          <li>Click "Choose file" to choose the files
            <code>C:\topmsv\data\st_1_file\st_1_ms1.msalign</code> and
            <code>C:\topmsv\data\st_2_ms2.msalign</code>, and click the "upload" button.</li>
          <p><img class="small-image" src="public/images/msalign_1.PNG" style="border: #00008B 1px solid;"/>
          </p>
          <li>Click "Choose file" to choose the file
            <code>C:\topmsv\data\st_1_file\st_1_frac.mzrt.csv</code>, and click "upload" button.</li>
          <p><img class="small-image" src="public/images/mzrt_1.PNG" style="border: #00008B 1px solid;"/></p>
        </ol>


        <h4 class="sectionTitle">3.4.3 Visualization of deconvolution results</h4> 
        <ol class="textList">
          <li>When deconvolution results are avaiable, theoretical isotopic
            envelopes will be added to 2D views of MS1 and MS/MS spectra. In
            addition, a table of deconvoluted monoisotopic masses will be added
            to the visualization webpage.</li>
          <p><img class="small-image" src="public/images/msalign_2.PNG" style="border: #00008B 1px solid;"/></p>
          <li>The file st_1_fract.mzrt.csv contains features reported from the
            LC-MS map. When feature information is available, features will be
            shown as dotted rectangles in the 3D view of the LC-MS map. When the mouse
            pointer hovers over a rectangle, information of the feature will be shown.</li>
          <p><img class="small-image" src="public/images/mzrt_2.PNG" style="border: #00008B 1px solid;"/></p>
        </ol>

        <h4 class="sectionTitle">3.5 Uploading spectral identfication results</h4> 
        <p>Spectral identification results can be added to the LC-MS/MS data by
        (1) running TopPIC at the server or (2) runing TopPIC locally and
        uploading identification result files to the server. When spectral
        identification results are available, identified
        proteoform-spectrum-matches of MS/MS spectra can be shown and inspected.</p>

        <h4 class="sectionTitle">3.5.1 Running TopPIC</h4> 
        <ol class="textList">
          <li>Click the "TopPIC" button in the top right corner of the
            visualization webpage.</li>
          <p><img class="small-image" src="public/images/toppic_1_part.png" style="border: #00008B 1px solid;"/></p>
          <li>Click the first "Browse" button to choose the protein database file
            <code>C:\topmsv\data\uniprot-st.fasta</code>, and upload it. If
            PTM charaterization is needed, 
            click the third "Browse" button to choose the text file 
            <code>C:\topmsv\data\common_mods.txt</code>, and upload it.
            Choose parameter settings of TopPIC and click the "Submit" button to
            submit a task to the server. 
            A manual of parameter settings of TopPIC can be found 
            <a href="https://www.toppic.org/software/toppic/manual.html">here</a>.
          </li>
          <p><img class="small-image" src="public/images/toppic_2.png" style="border: #00008B 1px solid;"/></p>
          <li>After the task is submitted, the progress of the task can be found
            in the "Tasks" tab. When the task is finished, spectral
            identification results reported by TopPIC will be added to the
            database of the data set. 
          </li>
        </ol>
        <h4 class="sectionTitle">3.5.2 Uploading spectral identification results</h4> 
        <ol class="textList">
          <li>Run TopPIC to perform spectral identification of the data file
            st_1_ms2.msalign reported by TopFD. A tutorial for runing TopPIC can be found 
            <a href="https://www.toppic.org/software/toppic/tutorial.html">here</a>.
            Spectral identification result files reported by TopPIC include:
            <ul>
              <li>C:\topmsv\data\st_1_ms2_toppic_prsm_single.tsv:
                proteoform-spectrum-matches (PrSMs) reported from the file
                st_1_ms2.msalign. 
            </ul>
          </li>  
          <li>Click the "Upload" button in the top right corner of the
            visualization webpage.</li>
          <p><img class="small-image" src="public/images/manual_1_part.png" style="border: #00008B 1px solid;"/></p>
          <li>Click the "Choose file" button to select the file
            <code>C:\topmsv\data\st_1_ms2_toppic_prsm_single.tsv</code> and
            upload it.</li>
          <p><img class="small-image" src="public/images/prsm_1.PNG" style="border: #00008B 1px solid;"/></p>
        </ol>
        <h4 class="sectionTitle">3.5.3 Inspecting PrSMs</h4> 
        PrSMs identified by TopPIC can be shown and inspected.  
        <ol class="textList">
          <li>In the visualization webpage of an MS/MS spectrum with an
            identified PrSM, click the "Inspect" button to inspect the PrSM.</li>
          <p><img class="small-image" src="public/images/inspect_1.PNG" style="border: #00008B 1px solid;"/></p>
          <li>In the inspection webpage, the user can edit the protein sequence
            and then click the "Submit" button to inspect the match between the
            spectrum and the protein sequence.</li>
          <p><img class="small-image" src="public/images/inspect_2.PNG" style="border: #00008B 1px solid;"/></p>
          <li>After the "Submit" button is clicked, a 2D view of the spectrum, a
            2D view of the monoisotopic masses, and a table of matched masses
            will be shown. A manual for the operations of the views can be found at the 
            <a href="manual.html">manual</a> webpage.
          <p><img class="small-image" src="public/images/inspect_3.PNG" style="border: #00008B 1px solid;"/></p>
        </ol>
        </ol>
    </div>
  </div>
</div>

<script src="lib/jquery-1.7.1.min.js"></script>
<script>
    //adding a button to copy command line commands
    function copyToClipboard(cmdId){
        const copyText = document.getElementById(cmdId).textContent;
        const textArea = document.createElement('textarea');

        textArea.textContent = copyText;
        document.body.append(textArea);
        textArea.select();
        document.execCommand("copy");

        //show tooltip that shows the text is copied
        initializeTooltip(cmdId);

        //remove the text area
        removeTextArea(textArea);

    }
    function removeTextArea(textArea){//removing the textarea that was created to just for copying purpose
        textArea.remove();
    }

    function initializeTooltip(cmdId){
      const tooltip = document.createElement('div');

      document.getElementById(cmdId + "Container").appendChild(tooltip);
      tooltip.classList.add("copyBtnTooltip");
      tooltip.textContent = "Copied to clipboard";

      setTimeout(function resetToolTip(){//give time for a tooltip to appear and then completely disappear
        tooltip.style.opacity = "0";
        tooltip.style.transition = "opacity 1s";

        setTimeout(function removeToolTip(){
                tooltip.remove();
        }, 500);
      }, 200);
    }
</script>
</body>
</html>
